### ISSUE TITLE
debug plugin doesn't work on my Mac

### ISSUE THREAD

**Issue by JustinCao86** [geany/geany-plugins#1019]
![image](https://user-images.githubusercontent.com/53166073/96951345-d9127600-151e-11eb-9528-baa61112ae57.png)
As screenshot shown above, all the debug buttons are NOT available.
I just downloaded geany-1.36_osx-2.dmg from official website, and my Mac version is 10.15.7 Catalina. 
I also have GDB 9.2 installed from MacPorts at /opt/local/bin, codesigned, working fine from terminal.
Here's the Debug Messages:

`
10:40:43: Geany INFO		: Geany 1.36, C
10:40:43: Geany INFO		: GTK 3.24.14, GLib 2.62.4
10:40:43: Geany INFO		: System data dir: /Applications/Geany.app/Contents/Resources/share/geany
10:40:43: Geany INFO		: User config dir: /Users/Justin/.config/geany
10:40:43: GLib-GIO DEBUG	: _g_io_module_get_default: Found default implementation local (GLocalVfs) for ‘gio-vfs’
10:40:44: Geany INFO		: Loaded GTK+ CSS theme '/Applications/Geany.app/Contents/Resources/share/geany/geany.css'
10:40:44: Geany INFO		: Loaded GTK+ CSS theme '/Applications/Geany.app/Contents/Resources/share/geany/geany-3.20.css'
10:40:44: Geany INFO		: System plugin path: /Applications/Geany.app/Contents/Resources/lib/geany
10:40:44: Geany INFO		: Added filetype Clojure (61).
10:40:44: Geany INFO		: Added filetype Swift (62).
10:40:44: Geany INFO		: Added filetype Graphviz (63).
10:40:44: Geany INFO		: Added filetype Nim (64).
10:40:44: Geany INFO		: Added filetype Arduino (65).
10:40:44: Geany INFO		: Added filetype Genie (66).
10:40:44: Geany INFO		: Added filetype Cython (67).
10:40:44: Geany INFO		: Added filetype TypeScript (68).
10:40:44: Geany INFO		: Added filetype Groovy (69).
10:40:44: Geany INFO		: Added filetype Kotlin (70).
10:40:44: Geany INFO		: Added filetype Scala (71).
10:40:44: Geany INFO		: Added filetype JSON (72).
10:40:44: Geany INFO		: Added filetype CUDA (73).
10:40:44: Geany INFO		: Loaded libvte from /Applications/Geany.app/Contents/Resources/lib/libvte-2.91.0.dylib
10:40:44: Geany INFO		: Loaded:   /Applications/Geany.app/Contents/Resources/lib/geany/debugger.so (Debugger)
10:40:44: Geany INFO		: /Users/Justin/projects/TestGeany/cgnew.f : F77 (UTF-8)
10:40:44: Geany INFO		: /Users/Justin/projects/TestGeany/main.f90 : Fortran (UTF-8)
10:40:44: Geany INFO		: /Users/Justin/projects/TestGeany/modParamt.f90 : Fortran (UTF-8)
10:40:51: GLib DEBUG	: posix_spawn avoided (workdir specified) (fd close requested) 
10:42:46: GLib DEBUG	: posix_spawn avoided (automatic reaping requested) (fd close requested) 

`
It would be appreciated if someone could point out why this plugin doesn't work, or maybe there's some trick settings I should do first?

**Comment by elextr** [comment]
What actually "doesn't work", clearly from your screenshot the plugin runs?  Did you try something debuggerish, like setting a breakpoint and running the program to see if it stops?

**Comment by JustinCao86** [comment]
> What actually "doesn't work", clearly from your screenshot the plugin runs? Did you try something debuggerish, like setting a breakpoint and running the program to see if it stops?

![image](https://user-images.githubusercontent.com/53166073/96952977-9fdc0500-1522-11eb-95b1-b379eff383a0.png)
Thank you for replying. Here's the running screenshot, compile's fine, build's fine, and execute's fine, but how to start debug???
I've seen someone's instruction that points out the way to use this debug plugin, and that instruction clearly has those debug buttons available, but mine's not, that's the confusing part. (shown below)
![image](https://user-images.githubusercontent.com/53166073/96953268-3f999300-1523-11eb-9a56-47cae84f3c6b.png)
As I'm new to Geany, I doubt if there're some settings that I haven't done yet......

**Comment by elextr** [comment]
I don't use the debugger plugin, but I'd guess that the buttons are not available because the debugger isn't running, try the lower right button.

**Comment by codebrainz** [comment]
It's possible those icons are like that because your icon theme doesn't have the icons the Debugger plugin uses. You might try changing icon themes to some robust one like perhaps the default icon theme of your desktop environment.

**Comment by JustinCao86** [comment]
> I don't use the debugger plugin, but I'd guess that the buttons are not available because the debugger isn't running, try the lower right button.

Yeah, that was my guess too, the debugger doesn't load/run properly. The lower right button doesn't have much help, just one option, like this

![image](https://user-images.githubusercontent.com/53166073/96956963-00bc0b00-152c-11eb-85b4-4ac30ea5cc36.png)

**Comment by JustinCao86** [comment]
> It's possible those icons are like that because your icon theme doesn't have the icons the Debugger plugin uses. You might try changing icon themes to some robust one like perhaps the default icon theme of your desktop environment.

Not like that actually. The "Run" button just doesn't work.

**Comment by JustinCao86** [comment]
Sorry guys, I forgot to put the error message which popup when I try to "run" debug:
![image](https://user-images.githubusercontent.com/53166073/96957159-7922cc00-152c-11eb-9547-1fe606b326e4.png)

**Comment by codebrainz** [comment]
@JustinCao86 just to confirm, do you have GDB installed? If so, maybe try putting `gdb` in that "Debugger" text box?

**Comment by JustinCao86** [comment]
> @JustinCao86 just to confirm, do you have GDB installed? If so, maybe try putting `gdb` in that "Debugger" text box?

of course:
![image](https://user-images.githubusercontent.com/53166073/96958224-69f14d80-152f-11eb-9880-83c7e262907c.png)
and that text box is a none editable box -_-

**Comment by elextr** [comment]
Just because it is in your path in a terminal doesn't mean its in the path inherited by Geany.  If started from menus it may not read the same shell startup files.

**Comment by JustinCao86** [comment]
> Just because it is in your path in a terminal doesn't mean its in the path inherited by Geany. If started from menus it may not read the same shell startup files.

that's right, and I also tried this, I made a link to ensure gdb could be run from /usr/local/bin, and maybe, as what you said, there are some tricks about this "inherited" path used by Geany?

**Comment by frlan** [comment]
I think it's an PATH-issue on Mac: Running a short shell script from within Geany gives me (at my box)

`PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/usr/local/MacGPG2/bin:/Applications/Wireshark.app/Contents/MacOS:/usr/local/sbin`

**Comment by FunnyGiraffe** [comment]
do you fix it?
